src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/IndexOfAny4Values.cs

```diff
@@ -27,6 +27,13 @@ internal override T[] GetValues()
             return new[] { Unsafe.As<TImpl, T>(ref e0), Unsafe.As<TImpl, T>(ref e1), Unsafe.As<TImpl, T>(ref e2), Unsafe.As<TImpl, T>(ref e3) };
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal override bool ContainsCore(T value) =>
+            Unsafe.As<T, TImpl>(ref value) == _e0 ||
+            Unsafe.As<T, TImpl>(ref value) == _e1 ||
+            Unsafe.As<T, TImpl>(ref value) == _e2 ||
+            Unsafe.As<T, TImpl>(ref value) == _e3;
```

Comment:
Can these use normal pointer casting (making the method unsafe and suppressing the warning) instead of Unsafe.As?